### DISCUSSION TITLE
Pushing 1 bit image

### DISCUSSION THREAD

**Discussion by DjordjeMandic** [Bodmer/TFT_eSPI#2719] [Q&A - General] [answered]
I got monochrome BMP that needs to be printed on screen with minimal flash and ram usage. 

The way how i did it:

1) Created 1 bit 32x32 BMP in photoshop which resulted in 192 bytes file.
2) Uploaded that file to [https://mischianti.org/images-to-byte-array-online-converter-cpp-arduino/](https://mischianti.org/images-to-byte-array-online-converter-cpp-arduino/)
3) Selected `Arduino Code` as output format with `Add size to array` checked. `Draw Mode` set to `Horizontal - 1 bit per pixel`
4) Copied the result `const unsigned char imageArray [128] PROGMEM = {.....];` and used it with function `sprite.pushImage(0,0,32,32, (uint16_t*)imageArray, 1);`

It works but im interested is there a better way to do it?

Whole point of doing it as 1 bit is being able to create monochrome icons with multiple color combiantions (changing bitmap color before pushing sprite)

**Comment by samwh1te** [comment]
Yes!! What would be the optimised way to push a monochrome image to a display?

**Reply by samwh1te** [reply, marked as answer]
XBM. The Generic/drawXBitmap example
